
# Compared Methods  

Our current code includes implementations of comparative methods such as **Tangram**, **gimVI**, **SpaGE**, **stPlus**, and **spatialscope**.  


## Data Sources and Code Availability  

### 1. Methods from the *SpatialBenchmarking* Paper  
The code and data for the following methods can be found in the paper:  
**《Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution》**  
(https://github.com/QuKunLab/SpatialBenchmarking), including:  
- **SpaGE**, **gimVI**, **novoSpaRc**, **stPlus**, **LIGER**, **Seurat**, **SpaOTsc**.  

### 2. Recent Methods from Official Repositories  
For newer methods, access their official GitHub repositories for code and data, and follow the `README` instructions to reproduce results:  
- **stDiff**: https://github.com/fdu-wangfeilab/stDiff
- **TISSUE**: https://github.com/sunericd/TISSUE  
- **SPRITE**: https://github.com/sunericd/SPRITE  
