
# Compared Methods  

Following the *SpatialBenchmarking* [[paper](https://github.com/QuKunLab/SpatialBenchmarking)], we can obtain the [[test_pipeline.py](./test_baseline.py)] for comparison. After running it, you will get the results of Tangram, SpaGE, gimVI, novoSpaRc, SpaOTsc, and stPlus. 

For the other four comparison methods, spscope can be obtained by running [[test_spscope.py](./test_spscope.py)], SPRITE can be obtained by running [[test_TISSUE.py](./test_TISSUE.py)], SPRITE can be obtained by running [[test_sprite.py](./test_sprite.py)], and stDiff can be obtained by running [[test_stDiff.py](./test_stDiff.py)].




## Data Sources and Code Availability  

### 1. Methods from the *SpatialBenchmarking* Paper  
The code and data for the following methods can be found in the paper:  
**《Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution》**  
(https://github.com/QuKunLab/SpatialBenchmarking).

### 2. Recent Methods from Official Repositories  
For newer methods, access their official GitHub repositories for code and data, and follow the `README` instructions to reproduce results:  
- **stDiff**: https://github.com/fdu-wangfeilab/stDiff
- **TISSUE**: https://github.com/sunericd/TISSUE  
- **SPRITE**: https://github.com/sunericd/SPRITE  
