
# Compared Methods  

Following the *SpatialBenchmarking* [[paper](https://github.com/QuKunLab/SpatialBenchmarking)], we can obtain the [[test_pipeline.py](./test_pipeline.py)] for comparison. After running it, you will get the results of Tangram, SpaGE, gimVI, novoSpaRc, SpaOTsc, and stPlus. 

For the other four comparison methods, spscope can be obtained by running [[test_spscope.py](./test_spscope.py)]. For the remaining three methods, you can go to the corresponding files and follow the instructions in the `README.md` for installation and inference.




## Data Sources and Code Availability  

### 1. Methods from the *SpatialBenchmarking* Paper  
The code and data for the following methods can be found in the paper:  
**《Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution》**  
(https://github.com/QuKunLab/SpatialBenchmarking), including:  
- **SpaGE**, **gimVI**, **novoSpaRc**, **stPlus**, **LIGER**, **Seurat**, **SpaOTsc**.  

### 2. Recent Methods from Official Repositories  
For newer methods, access their official GitHub repositories for code and data, and follow the `README` instructions to reproduce results:  
- **stDiff**: https://github.com/fdu-wangfeilab/stDiff
- **TISSUE**: https://github.com/sunericd/TISSUE  
- **SPRITE**: https://github.com/sunericd/SPRITE  
